packages/test/test-utils/src/timeoutUtils.ts

```diff
@@ -4,15 +4,120 @@
  */
 
 import { Container } from "@fluidframework/container-loader";
+import { assert } from "@fluidframework/common-utils";
 
+// @deprecated this value is no longer used
 export const defaultTimeoutDurationMs = 250;
 
+// TestTimeout class manage tracking of test timeout. It create a timer when timeout is in effect,
+// and provide a promise that will be reject before the test timeout happen with a `timeBuffer` of 15 ms.
+// Once rejected, a new TestTimeout object will be create for the timeout.
+
+type PromiseRejectFunc = (reason?: any) => void;
+const timeBuffer = 15; // leave 15 ms leeway for finish processing
+
+class TestTimeout {
+    private timeout: number = 0;
+    private timer: NodeJS.Timeout | undefined;
+    private reject: PromiseRejectFunc = () => { };
+    private readonly promise: Promise<void>;
+    private rejected = false;
+
+    private static instance: TestTimeout;
+    public static initialize() {
+        TestTimeout.instance = new TestTimeout();
+    }
+    public static reset(runnable: Mocha.Runnable) {
+        TestTimeout.clear();
+        TestTimeout.instance.resetTimer(runnable);
+    }
+
+    public static clear() {
+        if (TestTimeout.instance.rejected) {
```

Comment:
instance will be undefined if not in mocha. surprised this compiles. we must have strict null of in the test project

Reply:
strict null check is on, but it must be a hole in type script type detection.

Reply:
you should add ` | undefined` to the type declaration of `instance` and handle undefined

Reply:
I changed it to be always initialized.